app/connector/ftp/src/main/java/io/syndesis/connector/ftp/FtpConnectorCustomizer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.ftp;
+
+import java.util.Map;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.syndesis.common.util.ErrorCategory;
+import io.syndesis.common.util.SyndesisConnectorException;
+import io.syndesis.common.util.json.JsonUtils;
+import io.syndesis.integration.component.proxy.ComponentProxyComponent;
+import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
+
+public class FtpConnectorCustomizer implements ComponentProxyCustomizer {
+
+    static final String CAMEL_FILE_NAME = "CamelFileName";
+
+    @Override
+    public void customize(ComponentProxyComponent component, Map<String, Object> options) {
+
+        component.setBeforeProducer(FtpConnectorCustomizer::doBeforeProducer);
+
+        component.setBeforeConsumer(FtpConnectorCustomizer::doBeforeConsumer);
+        component.setAfterConsumer(FtpConnectorCustomizer::doAfterConsumer);
+    }
+
+    // Before Uploading or Updating a named file (pattern: to)
+    private static void doBeforeProducer(Exchange exchange) throws JsonMappingException, JsonProcessingException {
+        final Message in = exchange.getIn();
+        if (in.getBody(String.class) != null && JsonUtils.isJson(in.getBody(String.class))) {
+            final FtpPayload payLoad = new ObjectMapper().readValue(in.getBody(String.class), FtpPayload.class);
```

Comment:
invoking `Message:getBody` might involve type conversion, so it could be costly, perhaps inline this?

```suggestion
        final String body = in.getBody(String.class);
        if (body != null && JsonUtils.isJson(body)) {
            final FtpPayload payLoad = new ObjectMapper().readValue(body, FtpPayload.class);
```

Reply:
fixed